lib/genesis/helpers.go

```diff
@@ -82,3 +85,199 @@ func NewGenesisBlockFromTrie(t *trie.Trie) (*types.Header, error) {
+		return nil, err
+	}
+
+	g := new(Genesis)
+
+	err = json.Unmarshal(data, g)
+
+	grt := g.Genesis.Runtime
+	res := buildRawMap(grt)
+
+	g.Genesis.Raw[0] = res
+
+	return g, err
+}
+
+// keyValue struct to hold data regarding entry
+type keyValue struct {
+	key      []string
+	value    string
+	valueLen *big.Int
+}
+
+func buildRawMap(m map[string]map[string]interface{}) map[string]string {
+	res := make(map[string]string)
+	for k, v := range m {
+		kv := new(keyValue)
+		kv.key = append(kv.key, k)
+		buildRawMapInterface(v, kv)
+
+		key := formatKey(kv.key)
+
+		value, err := formatValue(kv)
+		if err != nil {
+			// todo determine how to handle error
+		}
+		res[key] = value
+	}
+	return res
+}
+
+func buildRawMapInterface(m map[string]interface{}, kv *keyValue) {
+	for k, v := range m {
+		kv.key = append(kv.key, k)
+		switch v2 := v.(type) {
+		case []interface{}:
+			kv.valueLen = big.NewInt(int64(len(v2)))
+			buildRawArrayInterface(v2, kv)
+		case string:
+			kv.value = v2
+		}
+	}
+}
+
+func buildRawArrayInterface(a []interface{}, kv *keyValue) {
+	for _, v := range a {
+		switch v2 := v.(type) {
+		case []interface{}:
+			buildRawArrayInterface(v2, kv)
+		case string:
+			// todo check to confirm it's an address
+			tba := crypto.PublicAddressToByteArray(common.Address(v2))
+			kv.value = kv.value + fmt.Sprintf("%x", tba)
+		case float64:
+			encVal, err := scale.Encode(uint64(v2))
+			if err != nil {
+				//todo determine how to handle this error
+			}
+			kv.value = kv.value + fmt.Sprintf("%x", encVal)
+		}
+	}
+}
+
+func formatKey(key []string) string {
+	switch true {
+	case reflect.DeepEqual([]string{"grandpa", "authorities"}, key):
+		kb := []byte(`:grandpa_authorities`)
+		return common.BytesToHex(kb)
+	case reflect.DeepEqual([]string{"system", "code"}, key):
+		kb := []byte(`:code`)
+		return common.BytesToHex(kb)
+	default:
+		var fKey string
+		for _, v := range key {
+			fKey = fKey + v + " "
+		}
+		fKey = strings.Trim(fKey, " ")
+		fKey = strings.Title(fKey)
+		kb := common.TwoxHash128([]byte(fKey))
+		return common.BytesToHex(kb)
+	}
+}
+
+func formatValue(kv *keyValue) (string, error) {
+	switch true {
+	case reflect.DeepEqual([]string{"grandpa", "authorities"}, kv.key):
+		if kv.valueLen != nil {
+			lenEnc, err := scale.Encode(kv.valueLen)
+			if err != nil {
+				return "", err
+			}
+			// prepend 01 to grandpa_authorities values
+			return fmt.Sprintf("0x01%x%v", lenEnc, kv.value), nil
+		}
+		return "", fmt.Errorf("error formatting value for grandpa authorities")
+	case reflect.DeepEqual([]string{"system", "code"}, kv.key):
+		return kv.value, nil
+	default:
+		if kv.valueLen != nil {
+			lenEnc, err := scale.Encode(kv.valueLen)
+			if err != nil {
+				return "", err
+			}
+			return fmt.Sprintf("0x%x%v", lenEnc, kv.value), nil
+		}
+		return fmt.Sprintf("0x%x", kv.value), nil
+	}
+}
+
+func BuildFromMap(m map[string][]byte, gen *Genesis) {
+	for k, v := range m {
+		fmt.Printf("key %x vL %v\n", k, len(v))
```

Comment:
Not sure why you are doing fmt in this file instead of log? On the other hand It seems like this is not yet finished? So, just adding it as reminder for you to not forget to cleanup. Same for other fmt I've seen on this file 👍